### ISSUE TITLE
No bandpass solutions above 1500 MHz

### ISSUE THREAD

**Issue by mbietenholz** [idia-astro/pipelines#46]
The IDIA pipeline (version 1) did not get any solutions in the bandpass above 1500 MHz on my dataset.

Attached is a plot of the bandpass solutions from the IDIA pipeline - plot was made without any selection in channel, freq. and with automatic axis limits - bp solutions seem to stop just below 1.5 GHz.  The calibrator is J0408 so it should be pretty strong, so I would expect some solutions 1.6 to 1.7 GHz.

(The files are in  /idia/users/mbietenholz/AT2019azh_2019nov25)

![1574640063_sdp_l0 bcal](https://user-images.githubusercontent.com/29401541/83786424-aa276080-a660-11ea-92ee-363004b5c7ff.png)

**Comment by Jordatious** [comment]
Hey @mbietenholz, 

In this run, you've used the following bad frequency ranges in your config, which appear to be the old defaults.

```
badfreqranges = [ '935~947MHz',   # List of bad frequency ranges (to flag)
        '1160~1310MHz',
        '1476~1611MHz',
        '1670~1700MHz']
```

The new defaults are better, and we're considering updating them further:

```
badfreqranges = [ '933~960MHz',   # List of bad frequency ranges (to flag)
                  '1163~1299MHz',
                  '1524~1630MHz']
```

Quite a lot of data above 1.5 GHz is flagged, but not all of it. Attached are the solutions derived for J0408-6545. There seems to be very little data above 1.5 GHz, as you say (as well as bad solutions at ~860-880 MHz, perhaps from RFI). In comparison, there's much more data in this range for J0842+1835. 

But I'm not immediately sure why no data at all is shown above 1.5 GHz in your plot. I presume you're showing corrected data. Have you chosen any averaging for this plot? It's possible there's a bunch of RFI within that range that has automatically been flagged. Can you plot the same data before flagging is performed?

@Kitchi, is there any reason the solutions would show data here, but not the corrected data?

![image](https://user-images.githubusercontent.com/10542136/83803466-e8e30800-a6ac-11ea-88d4-0d104b6e8f63.png)

**Comment by IanHeywood** [comment]
> The new defaults are better, and we're considering updating them further

@mbietenholz you can find these in this branch here:

https://github.com/idia-astro/pipelines/blob/selfcal_dev/processMeerKAT/default_config.txt#L53

> There seems to be very little data above 1.5 GHz, as you say

I think what's going on here is that the default badfreq ranges only leave that 59 MHz-wide strip at 1611 MHz, everything else above 1476 MHz is flagged. You can probably check the flagversions table to see whether the subsequent autoflagging step is taking what remains out also, but that region also looks very sparse in the visibility plot above (and the amplitude points are also below where you'd expect them to be given the source and the rest of the corrected spectrum).

Maybe the data there are generally ropey (I notice the new defaults partially covers this region).

> bad solutions at ~860-880 MHz, perhaps from RFI

I'm fairly sure this is just the roll-off at the edge of the bandpass. Doesn't look like the lower end is flagged explicitly.

> is there any reason the solutions would show data here, but not the corrected data?

Solution tables don't have a data/corrected distinction, @mbietenholz just plotted the amplitudes from the bandpass table.

Cheers.

**Comment by IanHeywood** [comment]
> we're considering updating them further

I think the hard frequency masks in the pipeline are a bit harsh. There's a "MeerKAT cookbook" notebook that has a good list to use. I found that flagging the band edges for all baselines, but then only applying that RFI mask list to spacings below 600 m works well. 

https://github.com/IanHeywood/oxkat/blob/master/oxkat/1GC_02_casa_basic_flags.py#L34

The RFI occupancy is much reduced on longer spacings, so you can use autoflaggers on what's left and derive a bandpass table that has much more complete coverage.

![bandpass](https://user-images.githubusercontent.com/6533799/83855593-35abfa80-a710-11ea-8160-87413e50b9bc.png)

Cheers.

**Comment by Jordatious** [comment]
Hi @mbietenholz and @IanHeywood,

Sorry, I assumed the plot you included was the one you sent via email recently (see below), which shows the corrected data, with averaging, which matches my plot above. I thought that maybe the few data points around 1.6 GHz might have disappeared with heavy averaging in plotms. I then quickly checked the plots written by the pipeline (`/idia/users/mbietenholz/AT2019azh_2019nov25/plots/`), and because they differed, I assumed they were the solutions! But obviously @mbietenholz's plot is solutions, and mine is corrected data. Nonetheless, I still wonder why there's corrected data near 1.6 GHz when there's no solutions there. I think it may be because of the default extrapolation behaviour we've previously discussed, which is now disabled. We now use `interp='linear,linearflag'` in all applycal calls.

Thanks for the input on the RFI mask @IanHeywood. That looks useful. I think @arusstaylor is investigating this now, so we will likely update the behaviour to what you're suggesting. I've also [updated](https://github.com/idia-astro/pipelines/blob/9f027a3462eadfae2e000ade8a3b8121485bcefc/processMeerKAT/default_config.txt#L47) the default SPW to start at 880 MHz, which can be considered as flagging everything below that. However, both the `spw` and `badfreqranges` are editable within any pipeline run, so if you want different values, you can use those.

@mbietenholz, I think running the current version of the pipeline from the `selfcal_dev` branch will show improved results. Please let us know how you go with that. Can we close this ticket now? 

![image](https://user-images.githubusercontent.com/10542136/83868821-d3f98980-a72b-11ea-96e8-b195b2a3abb6.png)

**Comment by Jordatious** [comment]
@mbietenholz, I'm going to close this ticket for now, as I believe it's resolved with the latest version. Please reopen it if you don't think it's resolved.
